### ISSUE TITLE
Early HMOVE glitch on some consoles

### ISSUE THREAD

**Issue by thrust26** [stella-emu/stella#1022]
There seems to be a different result of an early HMOVE on some consoles:

https://forums.atariage.com/topic/280660-hmove-variations

If we find out how this work, we should add this as an option in the Developer/TIA tab.

**Comment by ale-79** [comment]
Is there a rom available that is known to show the issue, and possibly a screenshot of the output from an affected console?

In the mentioned issue #699, I posted the HMOVE tables from my consoles ([link](https://github.com/stella-emu/stella/issues/699#issuecomment-692357606)). Those include 2 late units: a 7800 with a TIA manufactured in Oct '89 and a Junior with an Oct '90 one (and a 1991 copyright on the box, so basically from one of the final production runs of the console). In both cases there are no differences in positioning for cycle 73 and 74 HMOVEs.

Eckhard Stolberg's post quoted in the AA thread you linked, is referring to differences that occur with HMOVE _pulses_ in the visible part of the scanline, but, as @DirtyHairy says in his reply in that same thread, as far as we know, cycle 73 and 74 HMOVEs do NOT produce pulses in the visible part of the line because of the delay between the register strobe and the first pulse, and that a possible explanation would be that in some consoles that delay is shorter.
The result in testing my units in both cases (cycle 73 and 74), is that the first pulse occurs at the beginning of the Horizontal Blank. I attach here a table that I compiled when investigating HMOVE-RESxx race conditions, with the timing of the HMOVE pulses for HMOVE from cycle 73 to 3.
[HMOVE-RESxx_race_conditions.txt](https://github.com/stella-emu/stella/files/14584955/HMOVE-RESxx_race_conditions.txt)

So, are there confirmed cases of different timing of the HMOVE pulses?

**Comment by thrust26** [comment]
I am also not 100% sure. Bin [this](https://forums.atariage.com/topic/362744-moviecart-pal-artifacts/) thread the discussion came up again, so I figured we should at least notice.

**Comment by DirtyHairy** [comment]
Hm, if I play with the HMOVE delay I do not replice that issue at all, but instead completely break the kernel.

**Comment by thrust26** [comment]
There was a new finding lately. It seems that (for whatever reason) the players are not correctly positioned at the beginning. Something that we do not emulate correctly.

**Comment by thrust26** [comment]
More info: https://forums.atariage.com/topic/72308-hunchy-ii/?do=findComment&comment=903045

**Comment by DirtyHairy** [comment]
Yeah, I played with it a little, anything that systematically affects all late HMOVEs badly breaks the kernel.

**Comment by thrust26** [comment]
If you add this as an option to the TIA code, I will create the UI for it.

**Comment by thrust26** [comment]
The PlusCart menu also has problems on certain Jr console (the display is shifted more right each line). I am quite sure it uses early HMOVEs too. Seems like the early HMOVE doesn't move back by 8 but only by 7 pixel.

https://forums.atariage.com/topic/317039-pluscart-sd/?do=findComment&comment=5510832
